db/schema_tables.cc

```diff
@@ -1753,7 +1753,19 @@ mutation make_scylla_tables_mutation(schema_ptr table, api::timestamp_type times
     auto ckey = clustering_key::from_singular(*s, table->cf_name());
     mutation m(scylla_tables(), pkey);
     m.set_clustered_cell(ckey, "version", utils::UUID(table->version()), timestamp);
-    store_map(m, ckey, "cdc", timestamp, table->cdc_options().to_map());
+    auto cdc_options = table->cdc_options().to_map();
+    if (!cdc_options.empty()) {
+        store_map(m, ckey, "cdc", timestamp, cdc_options);
+    } else {
+        // Avoid storing anything for cdc disabled, so we don't end up with
```

Comment:
There's something I don't understand here. In previous patches, you made changes so that a "disabled" options get replaced by an empty map. But here you say that empty map is also not good enough - and if you see one, you replace it by a tombstone. So why do we need the changes in the previous patches? Why can't we just leave the map as it was and here check if it's set to "disabled" instead of checking it to be empty?

Reply:
You are seeing the evolution of the patchset here, this change came last in time.

It's possible to avoid the map-clearing patches and test for enabled directly. I think there is some (very) small merit in the code here having less understanding of the internals of cdc and just looking at whether the map is empty or not. But I'm not pretending the whole thing is an example of clean design and separation of concerns.